### PR TITLE
Fix supertibble label

### LINKED ISSUE #157: [BUG] rlang change to supertbl label
# Expected Behavior
As cited in #155, during testing of `make_labelled()`, a label assigned to supertibbles changed from `<redcap_supertbl>` to `<rdcp_spr>`.

https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/2d1b4f12a5aa3561fa032d69ed9ad15a3bab0ad5/utility/cli_message_examples_reprex.md?plain=1#L201-L209

## How to Reproduce the Bug:
It's expected this change took place during one of the `rlang` updates. An underlying component of `rlang::as_label()` is the `pillar` and `vctrs` packages, specifically [`vctrs::vec_ptype_full()`](https://vctrs.r-lib.org/reference/vec_ptype_full.html) which takes the first element of a class vector ("redcap_supertbl") and by default abbreviates it to 8 characters.

TBD if this is an issue we need to address or if it is acceptable as is.

### Checklist
Before submitting this issue, please check and verify below that the submission meets the below criteria:

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [ ] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR fixes a bug that was causing one of our error messages to print incorrectly. Since it required learning a bit about custom print methods I also added a customization to the header when our supertibbles print. Let me know if you think we should keep it.

# Proposed Changes 

- Added a `vec_ptype_abbr.redcap_supertbl()` method so that objects of the supertibble type get abbreviated to `suprtbl` in cli messages.
    - This is following advice from the [`vctrs` vignette](https://vctrs.r-lib.org/articles/pillar.html#fixing-the-data-type) which advises keeping labels to 6 characters because these labels also appear when objects are elements of a `tibble` or `list`
- Added a `tbl_sum.redcap_supertbl()` method so that printed supertibbles include a header with the object type and number of instruments
    - Check out [this vignette](https://pillar.r-lib.org/articles/extending.html) if you're interested in other customizations we could add!
- Added rerunning the cli examples to `utility/refresh.R` 

### Old

```r
missing_col_supertbl <- tibble(redcap_data = list()) %>%
  as_supertbl()
make_labelled(missing_col_supertbl)
#> Error in `make_labelled()`:
#> ✖ You've supplied `<rdcp_spr[,1]>` for `supertbl` which is not a valid
#>   value
#> ! Must contain `supertbl$redcap_metadata`
#> ℹ `supertbl` must be a REDCapTidieR supertibble, generated using
#>   `read_redcap()`
```

### New

```r
#> Error in `make_labelled()`:
#> ✖ You've supplied `<suprtbl[,1]>` for `supertbl` which is not a valid
#>   value
#> ! Must contain `supertbl$redcap_metadata`
#> ℹ `supertbl` must be a REDCapTidieR supertibble, generated using
#>   `read_redcap()`
```

### Old

```r
read_redcap(Sys.getenv("REDCAP_URI"), Sys.getenv("REDCAPTIDIER_CLASSIC_API")) %>%
  suppressWarnings()
#> # A tibble: 9 × 9
#>   redcap_form_name       redcap_form_label redcap_data redcap_metadata structure
#>   <chr>                  <chr>             <list>      <list>          <chr>    
#> 1 nonrepeated            Nonrepeated       <tibble>    <tibble>        nonrepea…
#> 2 nonrepeated2           Nonrepeated2      <tibble>    <tibble>        nonrepea…
#> 3 repeated               Repeated          <tibble>    <tibble>        repeating
#> 4 data_field_types       Data Field Types  <tibble>    <tibble>        nonrepea…
#> 5 text_input_validation… Text Input Valid… <tibble>    <tibble>        nonrepea…
#> 6 api_no_access          API No Access     <tibble>    <tibble>        nonrepea…
#> 7 api_no_access_2        API No Access 2   <tibble>    <tibble>        nonrepea…
#> 8 survey                 Survey            <tibble>    <tibble>        nonrepea…
#> 9 repeat_survey          Repeat Survey     <tibble>    <tibble>        repeating
#> # ℹ 4 more variables: data_rows <int>, data_cols <int>, data_size <lbstr_by>,
#> #   data_na_pct <formttbl>
```

### New

```r
read_redcap(Sys.getenv("REDCAP_URI"), Sys.getenv("REDCAPTIDIER_CLASSIC_API")) %>%
  suppressWarnings()
#> # A REDCapTidieR Supertibble with 9 instruments
#>   redcap_form_name       redcap_form_label redcap_data redcap_metadata structure
#>   <chr>                  <chr>             <list>      <list>          <chr>    
#> 1 nonrepeated            Nonrepeated       <tibble>    <tibble>        nonrepea…
#> 2 nonrepeated2           Nonrepeated2      <tibble>    <tibble>        nonrepea…
#> 3 repeated               Repeated          <tibble>    <tibble>        repeating
#> 4 data_field_types       Data Field Types  <tibble>    <tibble>        nonrepea…
#> 5 text_input_validation… Text Input Valid… <tibble>    <tibble>        nonrepea…
#> 6 api_no_access          API No Access     <tibble>    <tibble>        nonrepea…
#> 7 api_no_access_2        API No Access 2   <tibble>    <tibble>        nonrepea…
#> 8 survey                 Survey            <tibble>    <tibble>        nonrepea…
#> 9 repeat_survey          Repeat Survey     <tibble>    <tibble>        repeating
#> # ℹ 4 more variables: data_rows <int>, data_cols <int>, data_size <lbstr_by>,
#> #   data_na_pct <formttbl>
```

### Issue Addressed
Closes #157

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR
- [x] Pre-release package version incremented using `usethis::use_version()`

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

**Comment by skadauke** [comment]
I think this looks awesome! One tiny thing: please capitalize the final R in REDCapTidieR. :)

File: R/supertibble.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#' The object with `redcaptidier_supertbl` S3 class
+#'
+#' @keywords internal
+#'
+as_supertbl <- function(x) {
+  class(x) <- c("redcap_supertbl", class(x))
+  x
+}
+
+#' @export
```
Comment (rsh52) [inline R/supertibble.R:16]: Should we add an `@inherit` here in case anyone does `?REDCapTidieR:::vec_ptype_abbr.redcap_supertbl`

File: R/supertibble.R [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+vec_ptype_abbr.redcap_supertbl <- function(x) {
+  "suprtbl"
+}
+
+#' @export
```
Comment (rsh52) [inline R/supertibble.R:21]: Ditto above.

**Review by rsh52 — COMMENTED** [review]
Looks great to me, one minor comment about roxygen docs. I can't think of much else right now to apply this to, but it's good to know we can make use of it.

**Comment by rsh52** [comment]
Looks like the failure is a snapshot update which is expected.

The warnings weren't appearing for me but seem to be related to `.data` for a few instances I used in `clean_redcap_long()`. I think here:

https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/5ab694e76a7560e5aaebc0a7ef6237291f2a0581/R/clean_redcap_long.R#L87-L88

**Comment by ezraporter** [comment]
> I think this looks awesome! One tiny thing: please capitalize the final R in REDCapTidieR. :)

Good catch!

**Comment by ezraporter** [comment]
> Looks like the failure is a snapshot update which is expected.
> 
> The warnings weren't appearing for me but seem to be related to `.data` for a few instances I used in `clean_redcap_long()`. I think here:
> 
> https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/5ab694e76a7560e5aaebc0a7ef6237291f2a0581/R/clean_redcap_long.R#L87-L88

I think I fixed this in the last commit. I'll keep an eye on the CI. I went ahead and just upgraded everything in our `renv.lock` so package versions locally should be closer to what's running in CI.

**Review by rsh52 — APPROVED** [review]
